src/interface/variable_pack.hpp

```diff
@@ -70,9 +71,9 @@ class VariablePack {
  public:
   VariablePack() = default;
   VariablePack(const ViewOfParArrays<T> view, const ParArray1D<int> sparse_ids,
-               const std::array<int, 4> dims)
-      : v_(view), sparse_ids_(sparse_ids), dims_(dims),
-        ndim_((dims[2] > 1 ? 3 : (dims[1] > 1 ? 2 : 1))) {}
+               const ParArray1D<int> vector_component, const std::array<int, 4> dims)
```

Comment:
I'm pretty sure all these arguments can be made references so you aren't copying them twice.